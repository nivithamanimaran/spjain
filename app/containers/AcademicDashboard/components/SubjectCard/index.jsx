import React, { useState, useEffect } from 'react';
import {
  AppBar,
  Toolbar,
  Button,
  Grid,
  Typography,
  useTheme,
  Box,
  Table,
  TableBody,
  TableCell,
  TableHead,
  IconButton,
  TableRow,
  Accordion,
  LinearProgress,
  AccordionSummary,
  AccordionDetails,
  CircularProgress,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import KenCard from '../../../../global_components/KenCard';
// Icons
import { BiPencil } from 'react-icons/bi';
import { GrDocumentText } from 'react-icons/gr';
import { FiUsers } from 'react-icons/fi';
// Data
import AcademicData from '../../data.json';
import '../..//style.scss';

const useStyles = makeStyles(theme => ({
  gridBox: {
    padding: '10px',
    overflow: 'hidden',
    height: '81vh',
  },
  containerBox: {
    background: '#fff',
    width: '100%',
    height: '80vh',
    borderRadius: '9px !important',
  },
  title: {
    fontSize: '11px',
    fontWeight: 600,
  },
  toolbar: {
    padding: '0px',
    width: '100%',
    // borderBottom: '1px solid #D7DBE1',
    minHeight: '35px',
  },
  termBox: {
    background: '#FFF3DC',
    padding: '10px',
    textAlign: 'center',
  },
  sessionBox: {
    padding: '10px',
  },
  sessionHeader: {
    color: '#997AFF',
    fontSize: '13px',
    fontWeight: 600,
  },
  academicItems: {
    fontSize: '13px',
  },
  iconBase: {
    fontSize: '25px',
    position: 'relative',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
  },
  icons: {
    color: '#090F69',
    fontSize: '14px',
  },
  downloadIcon: {
    fontSize: '17px',
    position: 'relative',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
  },
  fileName: {
    fontSize: '13px',
    fontWeight: 600,
  },
  fileSubject: {
    fontSize: '11px',
    color: '#7A869A',
  },
  resource: {
    fontSize: '11px',
    color: '#7A869A',
    width: '100%',
    padding: '0px',
  },
  subjectCard: {
    padding: '10px !important',
  },
  iconBox: {
    height: '40px',
    width: '40px',
    margin: '5px',
  },
  subjectList: {
    overflowY: 'auto',
    '&::-webkit-scrollbar': {
      width: '4px',
    },
    '&::-webkit-scrollbar-track': {
      opacity: '0.2',
    },
    '&::-webkit-scrollbar-thumb': {
      backgroundColor: `#787878`,
      opacity: '0.1',
      outline: `1px solid #787878`,
    },
  },
  resourceList: {
    height: '67vh',
    overflowY: 'auto',
    '&::-webkit-scrollbar': {
      width: '4px',
    },
    '&::-webkit-scrollbar-track': {
      opacity: '0.2',
    },
    '&::-webkit-scrollbar-thumb': {
      backgroundColor: `#787878`,
      opacity: '0.1',
      outline: `1px solid #787878`,
    },
  },
  subjectName: {
    overflow: 'hidden',
    textOverflow: 'ellipsis',
  },
  star: {
    textAlign: 'right',
    width: '100%',
    float: 'right',
    marginRight: '11px',
    fontSize: '17px',
  },
  headerLeft: {
    width: '100%',
    justifyContent: 'space-between',
  },
}));

const CircularProgressProp = props => {
  return (
    <Box
      sx={{
        position: 'relative',
        display: 'inline-flex',
        width: '40px',
      }}
      style={props.module === 'dashboard' ? { left: '16%' } : { left: '0%' }}
    >
      <CircularProgress
        variant="determinate"
        value={props?.value}
        style={{ width: '100%', height: '100%' }}
      />
      <Box
        sx={{
          top: 0,
          left: 0,
          bottom: 0,
          right: 0,
          position: 'absolute',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Typography
          variant="caption"
          component="div"
          color="text.secondary"
        >{`${Math.round(props.value)}%`}</Typography>
      </Box>
    </Box>
  );
};
export default function SubjectCard(props) {
  const history = useHistory();
  const classes = useStyles();
  const theme = useTheme();
  const { t } = useTranslation();
  const { user, module } = props;
  const [loading, setLoading] = React.useState(false);
  const [subjectBar, setSubjectBar] = useState([]);

  React.useEffect(() => {
    const subjectBar = [];
    Object.entries(AcademicData['My Subjects']).map((item, keyVal) => {
      subjectBar.push({
        key: item[0],
        data: item[1],
      });
    });
    setSubjectBar(subjectBar);
  }, []);
  const handleViewSubject = () => {
    user?.Type === 'Student'
      ? history.push('/academics')
      : history.push('/facultyAcademics');
  };

  const showSubjectDetail = (e, item, index) => {
    console.log('e, item, index', e, item, index);
    history.push({
      pathname: `/class`,
      state: { data: subjectBar, selected: item },
    });
  };
  return (
    <Grid container xs={12}>
      <div className="container-card">
        <Grid container xs={12}>
          <Toolbar className={classes.toolbar}>
            <Grid item container xs={12} className={classes.headerLeft}>
              <Typography
                className={classes.title}
                style={{ fontSize: '14px', fontFamily: 'Work Sans' }}
              >
                {user?.Type === 'Student' ? 'MY SUBJECTS' : 'MY CLASSES'} (
                {subjectBar != undefined ? subjectBar?.length : 0})
              </Typography>
              <Typography
                style={{ fontSize: '13px', cursor: 'pointer' }}
                onClick={handleViewSubject}
              >
                View All
              </Typography>
            </Grid>
          </Toolbar>
        </Grid>
        <Grid
          container
          xs={12}
          className={classes.subjectList}
          style={
            module === 'dashboard' ? { height: '45vh' } : { height: '67vh' }
          }
        >
          {subjectBar.map((item, index) => {
            return (
              <Grid container xs={6} style={{ padding: '5px' }}>
                <div className="subjectCard">
                  <KenCard className="subject-card">
                    <Grid
                      xs={12}
                      style={{
                        height: '55px',
                        background: `${item.data.background}`,
                        padding: '10px',
                        marginBottom: '5px',
                        position: 'relative',
                        top: '50%',
                        left: '50%',
                        transform: 'translate(-50%, 0%)',
                        borderRadius: '9px',
                        textAlign: 'center',
                      }}
                      onClick={e => showSubjectDetail(e, item, index)}
                    >
                      {item.data['Subject Name']}
                    </Grid>
                    <Grid container xs={12}>
                      <Grid
                        xs={4}
                        style={{
                          overflow: 'hidden',
                          borderRight: '1px solid #d7dbe1',
                        }}
                      >
                        <CircularProgressProp
                          value={item.data.percentage}
                          module={module}
                        />
                      </Grid>
                      <Grid container xs={8} style={{ paddingLeft: '3px' }}>
                        <Grid xs={10}>
                          <Typography className={classes.resource}>
                            <GrDocumentText className={classes.icons} />{' '}
                            Resource:
                          </Typography>
                        </Grid>
                        <Grid xs={2}>
                          <Typography className={classes.resource}>
                            {item.data['resource']}{' '}
                          </Typography>
                        </Grid>
                        <Grid xs={10}>
                          <Typography className={classes.resource}>
                            <BiPencil className={classes.icons} /> Activities:
                          </Typography>
                        </Grid>
                        <Grid xs={2}>
                          <Typography className={classes.resource}>
                            {item.data['activities']}{' '}
                          </Typography>
                        </Grid>
                      </Grid>
                    </Grid>
                    {user?.Type != 'Student' ? (
                      <Grid
                        container
                        xs={12}
                        style={{
                          borderTop: '1px solid #D7DBE1',
                          paddingTop: '10px',
                        }}
                      >
                        <Grid xs={6}>{item.data.class}</Grid>
                        <Grid xs={6} style={{ textAlign: 'right' }}>
                          <span
                            style={{
                              background: '#D7DBE1',
                              padding: '3px',
                              borderRadius: '4px',
                            }}
                          >
                            <FiUsers /> <b>{item.data.NoOfStudents}</b>
                          </span>
                        </Grid>
                      </Grid>
                    ) : null}
                  </KenCard>
                </div>
              </Grid>
            );
          })}
        </Grid>
      </div>
    </Grid>
  );
}
