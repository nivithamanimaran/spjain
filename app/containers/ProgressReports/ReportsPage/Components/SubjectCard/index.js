import React, { useState } from 'react';
import { Grid, Typography, useTheme, Box } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import InfoIcon from '../../../../../assets/icons/InfoIcon.svg';
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
import RemoveCircleIcon from '@material-ui/icons/RemoveCircle';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import KenPopover from '../../../../../components/KenPopover/index';
import { useTranslation } from 'react-i18next';
import KenAvatar from '../../../../../components/KenAvatar/index';

const useStyles = makeStyles(theme => ({
  root: {
    // background: theme.palette.KenColors.gradeSectionSubjectBg,
    padding: 8,
    marginTop: 16,
    marginLeft: 4,
    marginBottom: 8,
    marginRight: 7,
    borderRadius: 6,
    // flexBasis: '32%',
    // [theme.breakpoints.down('xs')]: {
    //   flexBasis: '100%',
    // },
    border: `1px solid ${theme.palette.KenColors.neutral20}`,
  },

  subWrap: {
    justifyContent: 'space-between',
    alignItems: 'center',
    position: 'relative',
  },
  subValue: {
    alignItems: 'center',
    marginTop: 10,
    marginBottom: 10,
  },
  subRemarks: {
    alignItems: 'center',
  },
  subLabel: {
    color: theme.palette.KenColors.gradeTermLabel,
    fontSize: 12,
  },
  subRemarkLabel: {
    color: theme.palette.KenColors.gradeTermLabel,
    fontSize: 12,
    marginTop: 6,
    marginLeft: 4,
  },
  NoteIcon: {
    Width: 12,
    Height: 12,
    cursor: 'pointer',
  },
  gradeValue: {
    display: 'flex',
    textAlign: 'center',
    justifyContent: 'center',
    fontSize: 14,
  },
  gradeValueLabel: {
    width: 32,
    height: 32,
    borderRadius: '50%',
    fontWeight: 600,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: 14,
  },
  subjectMark: {
    marginLeft: 12,
    color: theme.palette.KenColors.gradeSectionHeaderLabel,
    fontSize: 24,
    fontWeight: 700,
  },
  upArrow: {
    backgroundColor: theme.palette.KenColors.a1BgColor,
    borderRadius: '50%',
    color: theme.palette.KenColors.tertiaryGreen504,
    height: '18px',
    padding: 2,
    width: '18px',
  },
  downArrow: {
    backgroundColor: theme.palette.KenColors.secondaryPeach2,
    borderRadius: '50%',
    color: theme.palette.KenColors.tertiaryRed501,
    padding: 2,
    height: '18px',
    width: '18px',
  },
  stableArrow: {
    backgroundColor: theme.palette.KenColors.tertiaryBlue100,
    borderRadius: '50%',
    color: theme.palette.KenColors.kenWhite,
    padding: 2,
    height: '18px',
    width: '18px',
  },
  forwardArrow: {
    backgroundColor: theme.palette.KenColors.tertiaryBlue100,
    borderRadius: '50%',
    color: theme.palette.KenColors.kenWhite,
    padding: 2,
    height: '18px',
    width: '18px',
  },
  infoIcon: {
    paddingRight: 4,
  },
  ratingsText: {
    color: theme.palette.KenColors.tertiaryBlue100,
  },
  markWrap: {
    // background: theme.palette.KenColors.neutral11,
    borderTopRightRadius: 6,
    borderTopLeftRadius: 6,
    border: `1px solid ${theme.palette.KenColors.gradeSectionBorder}`,
    width: 294,
  },
  markLabelColor: {
    color: theme.palette.KenColors.neutral100,
    fontSize: 14,
    fontWeight: 600,
    display: 'flex',
    alignItems: 'center',
    // height: 56,
    paddingTop: 16,
    paddingLeft: 16,
    // border: `1px solid ${theme.palette.KenColors.gradeSectionBorder}`,
  },
  markContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 10,
  },
  markLabel: {
    fontSize: 14,
    color: theme.palette.KenColors.tertiaryBlue504,
  },
  markValue: {
    fontSize: 14,
    fontWeight: 600,
    color: theme.palette.KenColors.tertiaryBlue504,
  },
  markItemContainer: {
    background: theme.palette.KenColors.kenWhite,
    padding: 16,
  },
  rollNumLabel: {
    color: theme.palette.KenColors.neutral400,
    fontSize: 12,
    marginLeft: 22,
  },
  infoIconWrap: {
    textAlign: 'end',
    height: 12,
  },
}));

export default function SubjectCard(props) {
  const classes = useStyles();
  const {
    name,
    rollNumber,
    subjectMark,
    subjectName,
    gradeValue,
    ratingsIcon,
    status,
    ratingsText,
    ratingsDifference,
    previousRatings,
    marksBreakdown = [],
    itemPerRow,
    fontSize,
    infoIcon,
    markLabel,
    variant,
  } = props;

  const divRef = React.useRef();
  const { t } = useTranslation();

  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const theme = useTheme();

  //getIcon based on status

  const getIcon = () => {
    if (ratingsIcon) return ratingsIcon;
    switch (status) {
      case 'stable':
        return (
          <RemoveCircleIcon fontSize="small" className={classes.stableArrow} />
        );

      case 'increased':
        return <ArrowUpwardIcon fontSize="small" className={classes.upArrow} />;

      case 'decreased':
        return (
          <ArrowDownwardIcon fontSize="small" className={classes.downArrow} />
        );

      default:
        return (
          <ArrowForwardIcon fontSize="small" className={classes.forwardArrow} />
        );
    }
  };

  // font color of grade

  const getGradeTextColor = () => {
    switch (gradeValue) {
      case 'A1':
      case 'A':
        return theme.palette.KenColors.a1FontColor;
      case 'A2':
        return theme.palette.KenColors.a2FontColor;
      case 'B1':
      case 'B':
        return theme.palette.KenColors.b1FontColor;
      case 'B2':
        return theme.palette.KenColors.b2FontColor;
      case 'C1':
      case 'C':
        return theme.palette.KenColors.c1FontColor;
      case 'C2':
        return theme.palette.KenColors.c2FontColor;
      case 'D':
        return theme.palette.KenColors.gradeDFontColor;
      case 'E':
        return theme.palette.KenColors.gradeEFontColor;

      default:
        return '';
    }
  };

  // background color of grade

  const getGradeBgColor = () => {
    switch (gradeValue) {
      case 'A1':
        return theme.palette.KenColors.a1BgColor;
      case 'A2':
        return theme.palette.KenColors.a2BgColor;
      case 'B1':
        return theme.palette.KenColors.b1BgColor;
      case 'B2':
        return theme.palette.KenColors.b2BgColor;
      case 'C1':
        return theme.palette.KenColors.c1BgColor;
      case 'C2':
        return theme.palette.KenColors.c2BgColor;
      case 'D':
        return theme.palette.KenColors.gradeDBgColor;
      case 'E':
        return theme.palette.KenColors.gradeEBgColor;

      default:
        return '';
    }
  };

  const handleInfoIconClick = () => {
    setAnchorEl(divRef.current);
  };

  return (
    <>
      <Grid className={classes.root} data-testid="subject-card">
        <Box className={classes.infoIconWrap}>
          {marksBreakdown.length > 0 && (
            <>
              <img
                src={InfoIcon}
                alt="infoIcon"
                className={classes.NoteIcon}
                onClick={handleInfoIconClick}
                ref={divRef}
              />
            </>
          )}
        </Box>
        <Grid container className={classes.subWrap}>
          <Grid container item alignItems="center">
            {variant === 'avatar' ? (
              <Typography style={{ marginLeft: 4, marginRight: 10 }}>
                <KenAvatar value={name.charAt(0)} />
              </Typography>
            ) : null}
            <Typography className={classes.subLabel} data-testid="subject-name">
              {name?.toUpperCase()}
            </Typography>

            <Typography
              className={classes.rollNumLabel}
              data-testid="subject-name"
            >
              {rollNumber}
            </Typography>
          </Grid>
        </Grid>

        <Grid container className={classes.subValue}>
          <Typography className={classes.gradeValue}>
            <Typography
              className={classes.gradeValueLabel}
              style={{
                color: getGradeTextColor(),
                backgroundColor: getGradeBgColor(),
                fontSize: fontSize,
              }}
              data-testid="grade-value"
            >
              {gradeValue}
            </Typography>
          </Typography>
          <Typography
            className={classes.subjectMark}
            data-testid="subject-mark"
          >
            {subjectMark}
          </Typography>
        </Grid>
        {/* <Grid container className={classes.subRemarks}>
          <Typography className={classes.infoIcon}>{getIcon()}</Typography>
          <Typography
            className={classes.subRemarkLabel}
            data-testid="subject-remark"
          >
            {ratingsDifference}{' '}
            <span
              className={classes.ratingsText}
              style={{ color: theme.palette.KenColors.tertiaryTeal502 }}
            >
              {ratingsText}
            </span>{' '}
            {previousRatings}
          </Typography>
        </Grid> */}
      </Grid>

      <KenPopover
        open={anchorEl}
        anchorEl={anchorEl}
        handleClose={() => {
          setAnchorEl(null);
        }}
        anchorOriginVertical="bottom"
        anchorOriginHorizontal="center"
        transformOriginVertical="top"
        transformOriginHorizontal="left"
      >
        <Grid className={classes.markWrap}>
          <Typography className={classes.markLabelColor}>
            {markLabel}
          </Typography>
          <Grid className={classes.markItemContainer}>
            {marksBreakdown?.map((item, index) => {
              return (
                <Grid key={index} className={classes.markContainer}>
                  <Typography className={classes.markLabel}>
                    {item.testItem}
                  </Typography>
                  <Typography className={classes.markValue}>
                    {item.testValue}
                  </Typography>
                </Grid>
              );
            })}
          </Grid>
        </Grid>
      </KenPopover>
    </>
  );
}
