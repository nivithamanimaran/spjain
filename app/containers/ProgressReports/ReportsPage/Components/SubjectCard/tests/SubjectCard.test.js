import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import SubjectCard from '../index';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Subject Card', () => {
  const div = document.createElement('div');
  it('Subject card renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <SubjectCard />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <SubjectCard />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('subject-card')).toMatchSnapshot();
  });

  it('Subject card renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <GradeSectionHeader
            name="English"
            subjectMark="90"
            gradeValue="A1"
            ratingsDifference="12.3"
            ratingsText="less than class avg of"
            previousRatings="80"
            status="increased"
          />
        </Router>
      </ThemeProvider>,
      div
    );

    const name = queryByTestId('subject-name');
    expect(name).toBeTruthy();
    expect(name.textContent).toBe('English');

    const subjectMark = queryByTestId('subject-mark');
    expect(subjectMark).toBeTruthy();
    expect(subjectMark.textContent).toBe('90');

    const gradeValue = queryByTestId('grade-value');
    expect(gradeValue).toBeTruthy();
    expect(gradeValue.textContent).toBe('A1');

    const ratingsDifference = queryByTestId('subject-remark');
    expect(ratingsDifference).toBeTruthy();
    expect(ratingsDifference.textContent).toBe('12.3');

    const ratingsText = queryByTestId('subject-remark');
    expect(ratingsText).toBeTruthy();
    expect(ratingsText.textContent).toBe('less than class avg of');

    const previousRatings = queryByTestId('subject-remark');
    expect(previousRatings).toBeTruthy();
    expect(previousRatings.textContent).toBe('80');
  });
});
