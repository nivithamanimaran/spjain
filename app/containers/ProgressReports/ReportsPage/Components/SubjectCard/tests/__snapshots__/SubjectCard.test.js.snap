// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subject Card Subject card renders correctly without props 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
  data-testid="subject-card"
>
  <div
    class="MuiGrid-root makeStyles-subWrap-2 MuiGrid-container"
  >
    <p
      class="MuiTypography-root makeStyles-subLabel-5 MuiTypography-body1"
    />
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      <img
        alt="infoIcon"
        class="makeStyles-NoteIcon-7"
        src="IMAGE_MOCK"
      />
    </p>
  </div>
  <div
    class="MuiGrid-root makeStyles-subValue-3 MuiGrid-container"
  >
    <p
      class="MuiTypography-root makeStyles-gradeValue-8 MuiTypography-body1"
    >
      <p
        class="MuiTypography-root makeStyles-gradeValueLabel-9 MuiTypography-body1"
        style="color: rgb(255, 255, 255); background-color: rgb(255, 255, 255);"
      />
    </p>
    <p
      class="MuiTypography-root makeStyles-subjectMark-10 MuiTypography-body1"
    />
  </div>
  <div
    class="MuiGrid-root makeStyles-subRemarks-4 MuiGrid-container"
  >
    <p
      class="MuiTypography-root makeStyles-infoIcon-15 MuiTypography-body1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-forwardArrow-14 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
        />
      </svg>
    </p>
    <p
      class="MuiTypography-root makeStyles-subRemarkLabel-6 MuiTypography-body1"
    >
       
      <span
        class="makeStyles-ratingsText-16"
        style="color: rgb(134, 147, 176);"
      />
       
    </p>
  </div>
</div>
`;
